### ISSUE TITLE
Lag opening emoji window

### ISSUE THREAD

**Issue by bergonzzi** [Neo11/ChitChat#19]
Adding this issue (https://github.com/stonesam92/ChitChat/issues/128) here as well so you can follow up when you have the time: 

Lately I experience some noticeable lag when I'm opening the emoji button. I already tried reinstalling but without any effect. Is this something also other users experience or not?

I'm using Chitchat on El Capitan / 10.11.3.

**Comment by c-o-m-m-a-n-d-e-r** [comment]
Yes, but this seems to be an bug in WhatsApp ... you can try http://web.whatsapp.com/ directly ... same behavior ...

**Comment by bergonzzi** [comment]
Yup indeed, it's true for Safari, although on Chrome and Firefox it seems to work pretty well.

Would it be possible or are there any disadvantages of using a Chrome container instead of Safari?

**Comment by c-o-m-m-a-n-d-e-r** [comment]
its not all the time lagging ...i think Apple / Whatsapp have to fix this ... 
I don't think you can change the webview to chrome / firefox that easy

**Comment by VagishVela** [comment]
I looked into this. The javascript that controls the opening/closing of the emoji bar is on line 6734 of the [main javascript file](https://web.whatsapp.com/main_f7c004e4b79068cee2be.js). If you search for `emojiExpanded` in there you will see the relevant pieces.

The click on the emoji button actually calls the javascript code which then inserts the code to show the emoji's, so that may be why there is a lag? It isn't a simple hide/show by modifying the css.

We could override the javascript for that part and make it open/close faster.

**Comment by bergonzzi** [comment]
Guys, are you also using macbook pros (retina screen) when testing this?
Just by chance I put the ChitChat window on my 2nd monitor (non-retina) and noticed that there's no lag when scrolling through the emojis. As soon as I put it back on the mbp screen, the images are reloaded (probably the @2x versions) and it starts lagging. I still wouldn't know how to solve it but it's another tip.

**Comment by c-o-m-m-a-n-d-e-r** [comment]
yep i use a retina mbp...

this lag is something whatsapp have to fix, cause it happens in webbrowser too!

**Comment by coreice** [comment]
I'm not sure what happened but the lag seems to be over. Sometimes a little lag with opening or changing emoji-categories but it responds faster then in the last months.

**Comment by ilker55** [comment]
I've noticed that the lag only happens on the build-in display on a MBP. On a secondary monitor it's smooth.
